interfaces/builtin/iio.go

```diff
@@ -113,7 +111,28 @@ func (iface *iioInterface) AppArmorConnectedPlug(spec *apparmor.Specification, p
 	deviceName := strings.TrimPrefix(path, "/dev/")
 	snippet = strings.Replace(snippet, "###IIO_DEVICE_NAME###", deviceName, -1)
 
+	// Add a snippet for various device specific rules, except for sysfs write
+	// access that are specialized below.
 	spec.AddSnippet(snippet)
+	// This part contains two sets of "**" that can use exponential memory when
+	// compiling without -O no-expr-simplify, so handle it with the parametric
+	// snippet workaround. Only one instance of this line will show up in the
+	// resulting profile. Eg, if the snap is assigned iio:device1 and
+	// iio:device3, this rule will expand to:
+	//  /sys/devices/**/{iio:device1,iio:device3}/** rwk,
+
+	// Because all deviceName values have the prefix of "iio:device" enforced
+	// by the sanitization logic above, we can trim that prefix and provide a
+	// shorter expansion expression.
```

Comment:
This comment seems misplaced now that we made similar changes to spi and i2c. Maybe move out to incorporate into the leading comment for AddParametricSnippet() and then here, in spi and i2c, just have:
```
// Use parametric snippets to avoid no-expr-simplify side-effects
```

(wordsmith as desired)

Reply:
Good idea, done now.